hbase-server/src/main/java/org/apache/hadoop/hbase/wal/WALSplitUtil.java

```diff
@@ -563,4 +564,58 @@ public int hashCode() {
 
     return mutations;
   }
+
+  /**
+   * Path to a file under recovered.hfiles directory of the region's column family: e.g.
+   * /hbase/some_table/2323432434/cf/recovered.hfiles/2332-wal. This method also ensures existence
```

Comment:
Why the 'wal' suffix? Is that right?

Reply:
Is the hfile named for the wal file that is being split?

Reply:
Same with the recovered edits. Let me copy the comment here.
// Append fileBeingSplit to prevent name conflict since we may have duplicate wal entries now.
// Append file name ends with RECOVERED_LOG_TMPFILE_SUFFIX to ensure
// region's replayRecoveredEdits will not delete it